substrate/primitives/runtime/src/lib.rs

```diff
@@ -1001,20 +1001,48 @@ macro_rules! assert_eq_error_rate_float {
 
 /// Simple blob to hold an extrinsic without committing to its format and ensure it is serialized
 /// correctly.
-#[derive(PartialEq, Eq, Clone, Default, Encode, Decode, DecodeWithMemTracking, TypeInfo)]
-pub struct OpaqueExtrinsic(Vec<u8>);
+#[derive(PartialEq, Eq, Clone, Default, Encode, Decode, DecodeWithMemTracking)]
+pub struct OpaqueExtrinsic(bytes::Bytes);
+
+impl TypeInfo for OpaqueExtrinsic {
+	type Identity = Self;
+	fn type_info() -> scale_info::Type {
+		scale_info::Type::builder()
+			.path(scale_info::Path::new("OpaqueExtrinsic", module_path!()))
+			.composite(
+				scale_info::build::Fields::unnamed()
+					.field(|f| f.ty::<Vec<u8>>().type_name("Vec<u8>")),
+			)
+	}
+}
 
 impl OpaqueExtrinsic {
+	/// Create a new instance of `OpaqueExtrinsic` from a `Vec<u8>`.
+	pub fn from_bytes(bytes: Vec<u8>) -> Self {
+		Self(bytes.into())
+	}
+
 	/// Convert an encoded extrinsic to an `OpaqueExtrinsic`.
-	pub fn from_bytes(mut bytes: &[u8]) -> Result<Self, codec::Error> {
```

Comment:
I hope people don't get confused and just replace usage of the old `from_bytes(something)` to the new `from_bytes(something.to_vec())` because it would be invalid.

Personally I would name the new method differently just to avoid the confusion. And better deprecate the old one rather than removing it IMO.

Reply:
Good point ! Done.